src/renderer/components/monaco-editor/monaco-editor.tsx

```diff
@@ -0,0 +1,290 @@
+    const uri = MonacoEditor.createUri(this.props.id ?? this.staticId);
+    const model = editor.getModels().find(model => model.uri.toString() == uri.toString());
+
+    if (model) {
+      return model; // already exists
+    }
+
+    const { language, value } = this.props;
+
+    return editor.createModel(value, language, uri);
+  }
+
+  @computed get options(): editor.IStandaloneEditorConstructionOptions {
+    return toJS({
+      ...UserStore.getInstance().editorConfiguration,
+      ...(this.props.options ?? {}),
+    });
+  }
+
+  /**
+   * Monitor editor's dom container element box-size and sync with monaco's dimensions
+   * @private
+   */
+  private bindResizeObserver() {
+    const resizeObserver = new ResizeObserver(entries => {
+      for (const entry of entries) {
+        const { width, height } = entry.contentRect;
+
+        this.setDimensions(width, height);
+      }
+    });
+
+    const containerElem = this.editor.getContainerDomNode();
+
+    resizeObserver.observe(containerElem);
+
+    return () => resizeObserver.unobserve(containerElem);
+  }
+
+  onModelChange = (model: editor.ITextModel, oldModel?: editor.ITextModel) => {
+    this.logger?.info("[MONACO]: model change", { model, oldModel });
+
+    this.saveViewState(oldModel);
+    this.editor.setModel(model);
+    this.restoreViewState(model);
+    this.editor.layout();
+    this.editor.focus(); // keep focus in editor, e.g. when clicking between dock-tabs
+    this.props.onModelChange?.(model, oldModel);
+    this.validateLazy();
+  };
+
+  /**
+   * Save current view-model state in the editor.
+   * This will allow restore cursor position, selected text, etc.
+   * @param {editor.ITextModel} model
+   */
+  saveViewState(model = this.model) {
+    if (!model) return;
+
+    MonacoEditor.viewStates.set(model.uri, this.editor.saveViewState());
+  }
+
+  restoreViewState(model = this.model) {
+    if (!model) return;
+
+    const viewState = MonacoEditor.viewStates.get(model.uri);
+
+    this.editor.restoreViewState(viewState);
+  }
+
+  componentDidMount() {
+    try {
+      this.createEditor();
+      this.logger?.info(`[MONACO]: editor did mount`);
+    } catch (error) {
+      this.logger?.error(`[MONACO]: mounting failed: ${error}`, this);
+    }
+  }
+
+  componentWillUnmount() {
+    this.unmounting = true;
+    this.saveViewState();
+    this.destroy();
+  }
+
+  private createEditor() {
+    if (!this.containerElem || this.editor || this.unmounting) {
+      return;
+    }
+    const { language, theme, readOnly, value: defaultValue } = this.props;
+
+    this.editor = editor.create(this.containerElem, {
+      model: this.model,
+      detectIndentation: false, // allow `option.tabSize` to use custom number of spaces for [Tab]
+      value: defaultValue,
+      language,
+      theme,
+      readOnly,
+      ...this.options,
+    });
+
+    this.logger?.info(`[MONACO]: editor created for language=${language}, theme=${theme}`);
```

Comment:
I think that having `id` would be useful as well in all of these log lines. Maybe having a getter for it too.

Reply:
Added computed `logMetadata: {id, model}` to every logger's output.